content/2016.02.localisation-service.md

```diff
@@ -0,0 +1,316 @@
+Title: The Mozilla Firefox localization dilema
+Lang: en
+Url: localization-service
+Slug: localization-service
+Date: 2016-02-10
+Summary: How to release Firefox in 90 languages and being able to fix mistakes.
+
+In this article, we present a proof-of-concept where Kinto is used to
+manage the translated strings of Firefox.
+
+# Problems we are trying to solve
+
+Under the [Go Faster project umbrella](https://wiki.mozilla.org/Firefox/Go_Faster>).
+which aims to ship new feature into Firefox faster, we investigate
+use-cases where our Kinto solution can bring value.
+
+It appeared that the Firefox localization process could gain a bit of
+flexibility.
+
+Today the release flow for Firefox Desktop is the following:
+
+- A release with the en-US built-in translation is made.
+- A string freeze for the release is triggered for the translators
+  community to translate the remaining missing strings.
+- As soon as a language is completely translated, the translators mark
+  it as ready.
+- The language translation strings are then review.
+- The translation strings are split into files (a file per view)
+- DTD Entities XML files for XUL based code and .properties string
+  bundles for JSM and C++ based code
+
+This flow put a lot of pressure on the contributors, who must ship
+translations of more than 90 languages as soon as Firefox string
+freeze is triggered.
+
+This process has to be repeated every 6 weeks for each Developer
+Edition, Beta, Release, ESR release channels and for all supported
+languages.
+
+To handle all the translation heavy flow, two main tools are used:
+
+- [Pontoon](https://pontoon.mozilla.org/) a UI to localize strings
+  in place.
+- [Elmo](https://l10n.mozilla.org) the central coordination
+  platform.
+
+If an error occurs in the flow, a chemspill release as to be built for
+the erroneous language. For typos, we usually have to wait the next
+train.
+
+
+# What are the solutions?
+
+Bundling Firefox like that takes too much time that we do not have. We
+are looking for ways to go faster and to paralize tasks.
+
+Ideally, we'd like to be able to localize during development, instead
+of waiting for string feeze.
+
+An alternate solution would be to rely on language packs updates via
+the addons channel.
+
+Thanks to the Nightly channel we have metrics on the frequency of
+displayed strings (from critical ones to rare security errors or
+deeply hidden menus).
```

Comment:
This actually isn't the case right now, sorry for misinformation.  There's been talk in Orlando about using telemetry for this task, but for now, we use hand-made `phases` lists as defined at https://wiki.mozilla.org/L10n:Pootle#Understanding_Phases